_posts/2023-01-18-#26697.md

```diff
@@ -12,13 +12,68 @@ commit:
+
+- The user can enable a logging category, for example `NET` (p2p messages) by specifing
+  the configuration (`bitcoin.conf` file or command-line) option `debug=net`.
+  The strings are defined by
+  [`LogCategories]`(https://github.com/bitcoin/bitcoin/blob/0c2a1288a210916254c939d080ce12ffc5547841/src/logging.cpp#L150)
+  but the mapping from the `enum` category symbols to strings is trivial.
+
+- The logging categories can also be enabled and disabled at runtime using the
+  [logging](https://github.com/bitcoin/bitcoin/blob/0c2a1288a210916254c939d080ce12ffc5547841/src/rpc/node.cpp#L231)
+  RPC.
+
+- Multiple logging categories can be enabled simultaneously by specifying
+  multiple `debug=` config options.
+
+- Some history (optional reading):
+  [PR 9424](https://github.com/bitcoin/bitcoin/pull/9424) changed the logging
+  categories from a set of strings (like `"net"`) to a enumeration type (`NET`).
+
 ## Questions
-1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)?
--->
+1. Did you review the PR?
+   [Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)?
+
+1. What is the problem that this PR proposes to address?
+
+1. What are the advantages of `std::bitset` compared with bit manipulations on integers?
+
+1. An earlier attempt, [PR 26619](https://github.com/bitcoin/bitcoin/pull/26619), was closed
+   in favor of the current PR. What are some advantages of each approach?
+   Do you agree with the decision to close 26619 in favor of 26697?
+
+1. There are two special logging categories, `ALL` and `NONE`. Why are these needed?
+
+1. [`Logger::m_categories`](https://github.com/bitcoin/bitcoin/pull/26697/files#diff-21abb6b14af1e9330a6f0c89a87231035a439248c556ef5e110eb0617b88a1f4L107)
+   was wrapped by `std::atomic`. Why was that needed? Why is it no longer needed?
+
+1. Following on to that question, some
+   [concern was expressed](https://github.com/bitcoin/bitcoin/pull/26697/files#r1054859149)
+   on the PR about performance, and the need for benchmarking. Why would this be important?
+
+1. An unknown P2P message does not cause a `LogPrintf()` (unconditional) message.
+   Wouldn't it be helpful to see if a peer is sending us a message we don't
+   understand? What's the design principle at work here?
+
+1. Does the same principle apply to RPC, REST, or ZMQ requests?
 
+1. C++ questions: The [`Logger::WillLogCategory`](https://github.com/bitcoin/bitcoin/pull/26697/files#diff-44fd50b51e8fc6799d38f193237fb921ec9d34306c448f64837524a17bac06eeL122)
+   is annotated with `const`; what does this mean?
+   How is it possible to modify the object by
+   [locking `m_cs`](https://github.com/bitcoin/bitcoin/pull/26697/files#diff-44fd50b51e8fc6799d38f193237fb921ec9d34306c448f64837524a17bac06eeR130)?
```

Comment:
As discussed in https://github.com/bitcoin-core-review-club/website/pull/588#pullrequestreview-1180735508 and #590, these are links to diffs in the current PR, so they will not persist if it changes. In the future, please use the review club repo for these types of links.